### PR TITLE
feat: incremental search and namespace preview in search bar

### LINKED ISSUE #606: Numaflow UI Search Bar Improvement
# Summary

What change needs making?
The current search bar needs the exact namespace name to find the result. Could we add, for example, incremental search?
Also it would be great if we can have a preview of all available namespaces.

# Use Cases

When would you use this?
I can enter partial name, for example, "test", to get all the available namespaces that have the word "test" in the name.
When I open the UI I can see a list of available namespaces and click the one I want from the list to enter the pipeline view.

### PR DESCRIPTION
**Pull request by veds-g**
Closes #606 

enhanced the namespace search bar to provide incremental search
provided a view of all available cluster namespaces on the search dropdown

image
image

### CODE REVIEW

File: server/apis/v1/handler.go [hunk 1]
```diff
@@ -322,6 +322,25 @@ func (h *handler) GetPipelineStatus(c *gin.Context) {
 	c.JSON(http.StatusOK, l)
 }
 
+// ListNamespaces is used to provide all the namespaces that have numaflow pipelines running
+func (h *handler) ListNamespaces(c *gin.Context) {
+	ns := c.GetString("namespace")
```
Comment (whynowy) [inline server/apis/v1/handler.go:327]: Do we need this?

File: server/apis/v1/handler.go [hunk 2]
```diff
@@ -322,6 +322,25 @@ func (h *handler) GetPipelineStatus(c *gin.Context) {
...
+	l, err := h.numaflowClient.Pipelines(ns).List(context.Background(), metav1.ListOptions{})
```
Comment (whynowy) [inline server/apis/v1/handler.go:328]: The `ns` here should be always empty string.

File: server/apis/v1/handler.go (hunk 1 above)
Comment (veds-g) [reply server/apis/v1/handler.go:327]: do you mean the api? it is to provide a preview of all available namespace to the user

File: server/apis/v1/handler.go (hunk 2 above)
Comment (veds-g) [reply server/apis/v1/handler.go:328]: right, it was the previous implementation i didn't modify it

**Review by whynowy — APPROVED** [review]

**Pull request merged**